File: app/src/main/java/android1601/itstep/org/kidsgame/program/adapters/ViewPagerAdapter.java
package android1601.itstep.org.kidsgame.program.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.ViewPager;

import java.util.ArrayList;
import java.util.List;

import android1601.itstep.org.kidsgame.program.data.Gifts;
import android1601.itstep.org.kidsgame.program.fragments.ToyDetailsFragment;

/**
 * Created by roman on 10.04.2017.
 */

public class ViewPagerAdapter extends FragmentStatePagerAdapter implements ViewPager.OnPageChangeListener {

    private static final String TAG = ViewPagerAdapter.class.getName();
    private List<Gifts> mItems;
    private ArrayList<ToyDetailsFragment> mFragments = new ArrayList<>();

    public ViewPagerAdapter(FragmentManager fm,ArrayList<Gifts>items) {
        super(fm);
        mItems = items;
        for (Gifts gifts : items){
            mFragments.add(null);
        }
    }


    @Override
    public Fragment getItem(int position) {
        ToyDetailsFragment toyDetailsFragment = ToyDetailsFragment.newInstance(mItems.get(position));
        mFragments.set(position,toyDetailsFragment);
        return toyDetailsFragment;
    }

    public int getItemCount(Object object){ return POSITION_NONE;}

    @Override
    public int getCount() {
        return mItems.size();
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

    }

    @Override
    public void onPageSelected(int position) {
        for (int i = 0; i < mFragments.size(); i++) {
            ToyDetailsFragment detailsFragment = mFragments.get(i);
            if (detailsFragment != null) {
                if (position == i) {
                    detailsFragment.startVoice();
                } else {
                    detailsFragment.stopAudio();
                }
            }
        }
    }

    @Override
    public void onPageScrollStateChanged(int state) {

    }
}

File: app/src/main/java/android1601/itstep/org/kidsgame/program/adapters/ViewPagerAdapter.kt
package android1601.itstep.org.kidsgame.program.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.ViewPager

import java.util.ArrayList

import android1601.itstep.org.kidsgame.program.data.Gifts
import android1601.itstep.org.kidsgame.program.fragments.ToyDetailsFragment
import androidx.viewpager.widget.PagerAdapter

/**
 * Created by roman on 10.04.2017.
 */

class ViewPagerAdapter(fm: FragmentManager, items: ArrayList<Gifts>) : FragmentStatePagerAdapter(fm), ViewPager.OnPageChangeListener {
    private val mItems: List<Gifts>
    private val mFragments = ArrayList<ToyDetailsFragment?>()

    init {
        mItems = items
        for (gifts in items) {
            mFragments.add(null)
        }
    }


    override fun getItem(position: Int): Fragment {
        val toyDetailsFragment = ToyDetailsFragment.newInstance(mItems[position])
        mFragments[position] = toyDetailsFragment
        return toyDetailsFragment
    }

    fun getItemCount(`object`: Any): Int {
        return PagerAdapter.POSITION_NONE
    }

    override fun getCount(): Int {
        return mItems.size
    }

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

    }

    override fun onPageSelected(position: Int) {
        for (i in mFragments.indices) {
            val detailsFragment = mFragments[i]
            if (detailsFragment != null) {
                if (position == i) {
                    detailsFragment.startVoice()
                } else {
                    detailsFragment.stopAudio()
                }
            }
        }
    }

    override fun onPageScrollStateChanged(state: Int) {

    }

    companion object {

        private val TAG = ViewPagerAdapter::class.java.name
    }
}
